Assets/BossRoom/Scripts/Client/Game/Character/ClientCharacter.cs

```diff
@@ -1,21 +1,40 @@
+using System;
 using UnityEngine;
 
 namespace BossRoom.Client
 {
-    [RequireComponent(typeof(BossRoom.NetworkCharacterState))]
+    [RequireComponent(typeof(NetworkCharacterState))]
     public class ClientCharacter : MLAPI.NetworkBehaviour
     {
         /// <summary>
-        /// The Vizualization GameObject isn't in the same transform hierarchy as the object, but it registers itself here
+        /// The Visualization GameObject isn't in the same transform hierarchy as the object, but it registers itself here
         /// so that the visual GameObject can be found from a NetworkObjectId.
         /// </summary>
-        public BossRoom.Visual.ClientCharacterVisualization ChildVizObject { get; set; }
+        public Visual.ClientCharacterVisualization ChildVizObject { get; private set; }
+
+        [Tooltip("PCs instantiate their graphics representation GameObject from this component.")]
+        [SerializeField]
+        CharacterContainer m_CharacterContainer;
```

Comment:
As discussed, since imps can be characters too, let's either have player character or avatar or something more specific to players

Reply:
This should be consistent throughout the project

Reply:
Refactored this to be part of `ClientAvatarGuidHandler` which exists only on Avatars (instead of this previous chunk of code residing on all Characters).